src/agent/transform-headers-agent.js

```diff
@@ -0,0 +1,81 @@
+/* eslint-disable no-underscore-dangle */
+const http = require('http');
+const WrappedAgent = require('./wrapped-agent');
+
+const { _storeHeader } = http.OutgoingMessage.prototype;
+
+class TransformHeadersAgent extends WrappedAgent {
+    transformRequest(request) {
+        const headers = {};
+        const hasConnection = request.hasHeader('connection');
+        const hasContentLength = request.hasHeader('content-length');
+        const hasTransferEncoding = request.hasHeader('transfer-encoding');
+        const hasTrailer = request.hasHeader('trailer');
+        const keys = request.getHeaderNames();
+
+        for (const key of keys) {
+            headers[key] = request.getHeader(key);
+            request.removeHeader(key);
+        }
+
+        if (!hasConnection) {
+            const shouldSendKeepAlive = request.shouldKeepAlive && (hasContentLength || request.useChunkedEncodingByDefault || request.agent);
+            if (shouldSendKeepAlive) {
+                headers[this.transformHeader('connection')] = 'keep-alive';
+            } else {
+                headers[this.transformHeader('connection')] = 'close';
+            }
+        }
+
+        if (!hasContentLength && !hasTransferEncoding) {
+            if (!hasTrailer && !request._removedContLen && typeof request._contentLength === 'number') {
+                headers[this.transformHeader('content-length')] = request._contentLength;
+            } else if (!request._removedTE) {
+                headers[this.transformHeader('transfer-encoding')] = 'chunked';
+            }
+        }
+
+        const sorted = Object.keys(headers)/* .sort(this.sort) */;
+
+        for (const key of sorted) {
+            request.setHeader(this.transformHeader(key), headers[key]);
+        }
+    }
+
+    addRequest(request, options) {
+        request._storeHeader = (...args) => {
```

Comment:
We need to wrap the internal `_storeHeader` function because it directly creates new headers (which we want to prevent, because we want to sort them first).